src/html5/node.rs

```diff
@@ -235,86 +237,96 @@ impl Clone for Node {
 
 impl Node {
     /// Create a new document node
+    #[must_use]
     pub fn new_document(document: &DocumentHandle) -> Self {
+        let (id, parent, children, name, namespace, is_registered) = <_>::default();
         Node {
-            id: Default::default(),
-            parent: None,
-            children: vec![],
+            id,
+            parent,
+            children,
             data: NodeData::Document(DocumentData::new()),
-            name: "".to_string(),
-            namespace: None,
+            name,
+            namespace,
             document: Document::clone(document),
-            is_registered: false,
+            is_registered,
```

Comment:
I looked into it, and there are some issue with the default implementation that I couldn't solve, for example I would have to use a default for the `NodeData` and neither of the variants looks like a good fit for a default value imo. Any suggestions?

Reply:
I guess, you could use a comment as a "Default" NodeData. But Document could also make sense, because it's an empty struct. I guess it's up to you, if no one as a other opinion on this.

Reply:
I would skip that for now, the PR is already big enough and I guess there might be more discussion about that, so I will open an Issue regarding this.